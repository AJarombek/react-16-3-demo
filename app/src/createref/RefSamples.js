/**
 * Example of a component that uses callback refs to focus upon an image.
 * @author Andrew Jarombek
 * @since 1/12/2020
 */

import React, {createRef} from 'react';
import { AJOutlinedButton } from 'jarombek-react-components';

import light from '../assets/light.png';

class RefSamples extends React.Component {

  constructor(props) {
    super(props);

    /* Callack Refs */
    this.lightFigure1 = null;

    this.setLightFigureRef = element => this.lightFigure1 = element;

    this.focusLightFigure1 = () => {
      if (this.lightFigure1) {
        this.lightFigure1.style.filter = 'brightness(5)';
      }
    };

    /* String Refs */
    this.focusLightFigure2 = () => {
      this.refs.lightFigure2.style.filter = 'brightness(5)';
    };

    /* createRef() Refs */
    this.lightFigure3 = createRef();

    this.focusLightFigure3 = () => this.lightFigure3.current.style.filter = 'brightness(5)';
  }

  render() {
    return (
      <div className="ref-samples">
        <div className="callback-ref-sample">
          <h3>Callback Ref</h3>
          <figure>
            <img src={light} ref={this.setLightFigureRef} />
          </figure>
          <AJOutlinedButton onClick={() => this.focusLightFigure1()} disabled={false}>
            Activate
          </AJOutlinedButton>
        </div>
        <div className="string-ref-sample">
          <h3>String Ref</h3>
          <figure>
            <img src={light} ref="lightFigure2" />
          </figure>
          <AJOutlinedButton onClick={() => this.focusLightFigure2()} disabled={false}>
            Activate
          </AJOutlinedButton>
        </div>
        <div className="createref-ref-sample">
          <h3>createRef() Ref</h3>
          <figure>
            <img src={light} ref={this.lightFigure3} />
          </figure>
          <AJOutlinedButton onClick={() => this.focusLightFigure3()} disabled={false}>
            Activate
          </AJOutlinedButton>
        </div>
      </div>
    );
  }
}

export default RefSamples;
