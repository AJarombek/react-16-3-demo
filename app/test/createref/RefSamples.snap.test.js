/**
 * Snapshot test for the {@link ../../src/createref/RefSamples} component.
 * @author Andrew Jarombek
 * @since 1/28/2020
 */

import React from 'react';
import RefSamples from '../../src/context/RefSamples';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<RefSamples />).toJSON();
  expect(tree).toMatchSnapshot();
});
