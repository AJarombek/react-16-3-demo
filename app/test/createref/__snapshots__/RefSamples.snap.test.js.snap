// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ref-samples"
>
  <div
    className="callback-ref-sample"
  >
    <h3>
      Callback Ref
    </h3>
    <figure>
      <img
        src="test-file-stub"
      />
    </figure>
    <div
      className="ajBaseButton-0-2-4 ajBaseButtonEnabled-0-2-5 aj-outlined-button ajOutlinedButton-0-2-1 ajOutlinedButtonEnabled-0-2-2"
      onClick={[Function]}
    >
      <button
        disabled={false}
        type="button"
      >
        Activate
      </button>
    </div>
  </div>
  <div
    className="string-ref-sample"
  >
    <h3>
      String Ref
    </h3>
    <figure>
      <img
        src="test-file-stub"
      />
    </figure>
    <div
      className="ajBaseButton-0-2-4 ajBaseButtonEnabled-0-2-5 aj-outlined-button ajOutlinedButton-0-2-1 ajOutlinedButtonEnabled-0-2-2"
      onClick={[Function]}
    >
      <button
        disabled={false}
        type="button"
      >
        Activate
      </button>
    </div>
  </div>
  <div
    className="createref-ref-sample"
  >
    <h3>
      createRef() Ref
    </h3>
    <figure>
      <img
        src="test-file-stub"
      />
    </figure>
    <div
      className="ajBaseButton-0-2-4 ajBaseButtonEnabled-0-2-5 aj-outlined-button ajOutlinedButton-0-2-1 ajOutlinedButtonEnabled-0-2-2"
      onClick={[Function]}
    >
      <button
        disabled={false}
        type="button"
      >
        Activate
      </button>
    </div>
  </div>
</div>
`;
